### PR TITLE
feat: add syntax for inline background color selection

### PR DESCRIPTION
**Pull request by shedric1**
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

New feature as discussed in #179 

Adds additional syntax to inline color selection so that both background and foreground colors can be customized in supported strings. Should be a non-breaking change. I have noticed no parsing errors or problems with existing themes. Previous in-line color syntax is only supplemented, not changed. As far as I know, no themes should need to be altered to keep working with the new feature.

Previous syntax for foreground color selection (unchanged):
`"prefix": "normal color string <#CB4B16>changed color string</>"`

New syntax for selecting both foreground and background:
`"prefix": "normal color string <#CB4B16,#000000>changed color string</>"`

Alternative syntax to change only background color, leaving foreground unchanged:
`"prefix": "normal color string <,#000000>changed color string</>"`
 
As shown, just separate the two colors by a comma to specify both. Or omit the first color and leave the comma to set only the background color

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by shedric1** [comment]
Reopening pull request after addressing issues with golangci code standards

File: renderer_test.go [hunk 1]
```diff
@@ -41,6 +41,27 @@ func TestWriteColorOverride(t *testing.T) {
...
+	renderer := &Renderer{
+		Buffer: new(bytes.Buffer),
+	}
+	text := "This is white, <,#000000>this is black</>, white again"
+	renderer.init("pwsh")
+	renderer.write("#193549", "#ff5733", text)
+	assert.NotContains(t, renderer.string(), "000000")
+}
+
+func TestWriteColorOverrideBoth(t *testing.T) {
```
Comment (JanDeDobbeleer) [inline renderer_test.go:66]: can you add a test with 16 colors eg: `"This is white, <black,white>this is black</>, white again"`

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

File: renderer_test.go (hunk 1 above)
Comment (shedric1) [reply renderer_test.go:66]: I added some additional unit tests to ensure the 16 color format is also properly handled, let me know if this is what you're looking for!

Comment (JanDeDobbeleer) [reply renderer_test.go:66]: I'm going to do a quick checkout of your branch as I feel the regex could be updated. But I'm not sure so I'll do a quick debug ;-)

Comment (JanDeDobbeleer) [reply renderer_test.go:66]: I'll refactor the regex later. It's out of scope for this one. Well done!

**Pull request merged**